modules/manifold/src/components/ui/segment-filters-control/segment-filters-control.js

```diff
@@ -0,0 +1,156 @@
+import {dotSet} from '../../../utils';
+import SegmentPanelList from './segment-panel-list';
+
+const Container = styled.div`
+  position: relative;
+`;
+
+const Header = styled.div`
+  position: absolute;
+  right: 0;
+  top: -28px;
+`;
+
+const Footer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding-bottom: 8px;
+`;
+
+const ConfirmButton = styled.div`
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+  color: ${props => (props.disabled ? '#aaa' : THEME_COLOR)};
+`;
+
+const IconButton = styled.div`
+  height: 24px;
+  padding-right: 8px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  margin: 1px;
+  transition: 0.2s;
+`;
+
+const _getSegmentFiltersStateFromProps = segmentFilters =>
+  segmentFilters.map((filters, segmentId) => ({
+    // _key is only use as keys to segment panels
+    _key: generateRandomId(),
+    _content: filters,
+  }));
+
+const _getSegmentFiltersPropsFromState = segmentFilters =>
+  segmentFilters.map(filters => filters._content);
+
+export default class SegmentFiltersControl extends PureComponent {
+  static propTypes = {
+    columnDefs: PropTypes.arrayOf(FIELD),
+    segmentFilters: STATE_DATA_TYPES.segmentFilters,
+    onUpdateSegmentFilters: PropTypes.func,
+  };
+
+  static defaultProps = {
+    columnDefs: [],
+    segmentFilters: [[], []],
+    onUpdateSegmentFilters: () => {},
+  };
+
+  // SegmentFiltersControl is reset to default state, controlled by a key, whenever baseCols change
+  state = {
+    segmentFilters: _getSegmentFiltersStateFromProps(this.props.segmentFilters),
+  };
+
+  _onUpdateSegmentFilters = () => {
+    this.setState({hasChanged: false}, () =>
+      this.props.onUpdateSegmentFilters(
+        _getSegmentFiltersPropsFromState(this.state.segmentFilters)
+      )
+    );
+  };
+
+  _onUpdateFilterValue = (segmentId, filterId, filterValue) => {
+    const {segmentFilters} = this.state;
+    const updatedSegmentFilters = dotSet(
+      segmentFilters,
+      [segmentId, '_content', filterId, 'value'],
+      filterValue
+    );
+    this.setState({
+      segmentFilters: updatedSegmentFilters,
+      hasChanged: true,
+    });
+  };
+
+  _onAppendSegment = () => {
+    const {segmentFilters} = this.state;
+    // just duplicate the last segment
+    const newSegment = segmentFilters[segmentFilters.length - 1]._content.map(
+      filter => ({
+        ...filter,
+        value: clone(filter.value),
+      })
+    );
+    this.setState({
+      segmentFilters: segmentFilters.concat({
+        _key: generateRandomId(),
+        _content: newSegment,
+      }),
+    });
+  };
+
+  _onRemoveSegment = segmentId => {
+    const {segmentFilters} = this.state;
+    if (segmentFilters.length > 2) {
+      this.setState({
+        segmentFilters: segmentFilters.filter((_, id) => id !== segmentId),
+        hasChanged: true,
+      });
+    }
```

Comment:
otherwise, show a message indicating we need at least two segments.

Reply:
In UI, there's already visual hint when you cannot click the delete button (button will be disabled). Adding error message requires some more complex logic so will add that later.